typescript/sdk/src/timelock/evm/EvmTimelockDeployer.hardhat-test.ts

```diff
@@ -164,4 +171,187 @@ describe('EvmTimelockDeployer', async () => {
+    assert(expectedConfig.cancellers, 'Expected cancellers to be defined');
+    for (const canceller of expectedConfig.cancellers) {
+      expect(await timelockInstance.hasRole(CANCELLER_ROLE, canceller)).to.be
+        .true;
+      if (!expectedConfig.proposers?.includes(canceller)) {
+        expect(await timelockInstance.hasRole(PROPOSER_ROLE, canceller)).to.be
+          .false;
+      }
+    }
+
+    // signer should not be the timelock admin after deployment
+    const timelockAdminRoleHash = await timelockInstance.TIMELOCK_ADMIN_ROLE();
+    expect(
+      await timelockInstance.hasRole(timelockAdminRoleHash, signer.address),
+    ).to.be.false;
+
+    // Timelock should still be the admin of itself
+    expect(
+      await timelockInstance.hasRole(
+        timelockAdminRoleHash,
+        timelockInstance.address,
+      ),
+    ).to.be.true;
+
+    // if an admin was set it should be the admin after the changes
+    if (expectedConfig.admin) {
+      expect(
+        await timelockInstance.hasRole(
+          timelockAdminRoleHash,
+          expectedConfig.admin,
+        ),
+      ).to.be.true;
+    }
+  }
+
+  it('should deploy with the correct canceller config', async () => {
+    const proposer = randomAddress();
+    const cancellerConfig: TimelockConfig = {
+      minimumDelay: 30,
+      proposers: [proposer],
+      executors: [signer.address],
+      cancellers: [randomAddress(), randomAddress()],
+    };
+
+    const { TimelockController } = await deployer.deployContracts(
+      TestChainName.test4,
+      cancellerConfig,
+    );
+    const timelockAddress = TimelockController.address;
+
+    const timelock = TimelockController__factory.connect(
+      timelockAddress,
+      multiProvider.getProvider(TestChainName.test4),
+    );
+
+    await assertCancellerConfig(cancellerConfig, timelock);
+  });
+
+  it('should not remove a proposer that it is also a canceller', async () => {
+    const proposer = randomAddress();
+    const cancellerConfig: TimelockConfig = {
+      minimumDelay: 30,
+      proposers: [proposer],
+      executors: [signer.address],
+      cancellers: [proposer, randomAddress()],
+    };
+
+    const { TimelockController } = await deployer.deployContracts(
+      TestChainName.test4,
+      cancellerConfig,
+    );
+    const timelockAddress = TimelockController.address;
+
+    const timelock = TimelockController__factory.connect(
+      timelockAddress,
+      multiProvider.getProvider(TestChainName.test4),
+    );
+
+    await assertCancellerConfig(cancellerConfig, timelock);
+  });
+
+  it('should set the expected admin from the config after applying the canceller changes', async () => {
+    const proposer = randomAddress();
+    const cancellerConfig: TimelockConfig = {
+      minimumDelay: 30,
+      proposers: [proposer],
+      executors: [signer.address],
+      cancellers: [proposer, randomAddress()],
+      admin: randomAddress(),
```

Comment:
Like my other comment, making the assertion through a shared function with a conditional inside that function makes these tests harder to reason about. You in effect have multiple dimensions you want to cross-test, so i would consider having multiple test groups with separate `beforeEach`s where you set the config, and then reusable, but explicit assertions about the on-chain state